infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -79,14 +81,26 @@ public InternalMachineConfig resolve() throws InfrastructureException {
         new InternalMachineConfig(
             null,
             toServers(containerEndpoints),
-            null,
+            toEnvVariables(wsAttributes),
             toMachineAttributes(pluginId, wsAttributes),
             toWorkspaceVolumes(cheContainer));
 
     normalizeMemory(container, machineConfig);
     return machineConfig;
   }
 
+  private Map<String,String> toEnvVariables(Map<String,String> wsAttributes) {
+    String envVars = wsAttributes
+        .get(format(SIDECAR_ENV_VARIABLES_ATTR_TEMPLATE, pluginPublisherAndName));
+    if (isNullOrEmpty(envVars)) {
+      return null;
+    }
+    return Stream.of(envVars.split(","))
```

Comment:
This breaks on `,` contained in the envvar value, right? IMHO that could render some `JAVA_OPTS` invalid, because e.g. settings for `-agentlib` or `-javaagent` are usually comma separated.

Reply:
Yeah, that is the problem. Also another example is LDAp namespaces like `myDN=OU=MY_OU,DC=MYDC,DC=local`